output/cloud/expv2/metrics_client.go

```diff
@@ -5,98 +5,69 @@ import (
 	"context"
 	"errors"
 	"fmt"
+	"io"
 	"net/http"
-	"sync"
-	"time"
+	"strings"
 
 	"github.com/klauspost/compress/snappy"
-	"github.com/sirupsen/logrus"
 	"google.golang.org/protobuf/proto"
 
 	"go.k6.io/k6/cloudapi"
-	"go.k6.io/k6/lib/consts"
 	"go.k6.io/k6/output/cloud/expv2/pbcloud"
 )
 
-type httpDoer interface {
-	Do(*http.Request) (*http.Response, error)
-}
-
 // metricsClient is a Protobuf over HTTP client for sending
 // the collected metrics from the Cloud output
 // to the remote service.
 type metricsClient struct {
-	httpClient httpDoer
-	logger     logrus.FieldLogger
-	token      string
-	userAgent  string
-
-	pushBufferPool sync.Pool
-	baseURL        string
+	httpClient *cloudapi.Client
+	url        string
 }
 
 // newMetricsClient creates and initializes a new MetricsClient.
-func newMetricsClient(logger logrus.FieldLogger, host string, token string) (*metricsClient, error) {
-	if host == "" {
-		return nil, errors.New("host is required")
+func newMetricsClient(c *cloudapi.Client, testRunID string) (*metricsClient, error) {
+	// Unfortunately, the cloudapi.Client works across different versions
+	// of the API, but it has the v1 harcoded so we need to trim the wrong path
+	// to be able to replace it with the correct one.
+	u := c.BaseURL()
```

Comment:
Maybe this should be done in the cloudapi package?

As a separate PR?

Reply:
I thought the same but I would avoid the load with a new PR to review for you folks. You're asking it so let's do it #3125